legacy-migration/providers/migrate.py

```diff
@@ -0,0 +1,312 @@
+        organization["Roles"] = [DEFAULT_ORGANIZATION_ROLE]
+        organizations.append(organization)
+    return organizations
+
+def migrate_administrators(provider_owner_guid, provider_id):
+    """Parses admin's fields between old-style format into new schema"""
+    all_the_groups_specific_provider = get_groups_by_provider(provider_owner_guid)
+    member_guids_specific_provider = get_all_provider_guids(all_the_groups_specific_provider)
+    # TODO: Iron out the pieces
+    if member_guids_specific_provider is None:
+        print('🚀 Could not retrieve the member guids for ', provider_id)
+    admin_usernames = get_admin_usernames(member_guids_specific_provider)
+    return admin_usernames
+
+def migrate_providers():
+    """Migrate the providers"""
+    provider_migration_count = 0
+    json_response = retrieve_legacy_providers()
+
+    # Bailout if request for providers failed
+    if json_response is None:
+        sys.exit("Failed to Retrieve any providers")
+
+    # provider_id_list = []
+
+    print('Beginning to Migrate' + str(len(json_response)) + ' Providers')
+
+    for provider in json_response:
+        provider_id = provider['provider']['provider_id']
+        print(f"Migrating {provider_id} providers")
+        description_of_holdings = provider['provider']['description_of_holdings']
+        
+        # Remove any "\" characters in the description of holdings string
+        # TODO item of discussion is the length that we will allow for DOH
+        # description_of_holdings=description_of_holdings.replace("\\","")
+        re.sub('[^A-Za-z0-9]+', '', description_of_holdings)
+        # print('Filtered description of holdings', description_of_holdings)
+        search_urls = provider['provider']['discovery_urls']
+        org_name = provider['provider']['organization_name']
+        contacts = provider['provider']['contacts']
+        provider_owner_guid = provider['provider']['id']
+
+        contact_persons = migrate_contact_persons(contacts)
+        
+        # Handle organizations
+        organizations = migrate_organizations(search_urls,provider_id,org_name)
+
+        # Handle the Administrators
+        admin_usernames = migrate_administrators(provider_owner_guid,provider_id)
+
+        # TODO A unit of discussion for how to handle cases where admin users is not provided 
+        if len(admin_usernames) < 1:
+            admin_usernames.append("defaultAdmin")
+
+    # Create the new record to be ingested by the new CMR provider interface
+        data = {}
+        data["MetadataSpecification"]= metadata_specification
+        data["ProviderId"] = provider_id
+        data["DescriptionOfHolding"] = description_of_holdings
+        data["Organizations"] = organizations
+        data["Administrators"] = admin_usernames
+        data["ContactPersons"] = contact_persons
+
+        # Dump the new metadata schema
+        json_data = json.dumps(data)
```

Comment:
could consider dumping with some indentation to improve readability
```python
json.dumps(data, indent=4)
```
https://stackoverflow.com/questions/12943819/how-to-prettyprint-a-json-file